### PR TITLE
Feat/onnx import tanh

### PR DESCRIPTION
**Pull request by AuruTus**
### Checklist

- [x] Confirm that `run-checks` script has been executed.

### Related Issues/PRs

Solve tanh task #600 

### Changes

add new unary node `tanh` to import ONNX model.

### Testing

add new test entry under `tests/onnx_tests.rs` and pass.

### CODE REVIEW

File: burn-import/onnx-tests/tests/tanh/tanh.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+class Model(nn.Module):
+    def __init__(self):
+        super(Model, self).__init__()
+
+    def forward(self, x):
+        return torch.tanh(x)
+
+
+def main():
+
```
Comment (antimora) [inline burn-import/onnx-tests/tests/tanh/tanh.py:18]: Can we add rand.seed for reproducibility like this:

```python
    # Set random seed for reproducibility
    torch.manual_seed(0)
```

File: burn-tensor/src/tensor/activation/base.rs [hunk 2]
```diff
@@ -69,3 +69,8 @@ pub fn log_sigmoid<const D: usize, B: Backend>(tensor: Tensor<B, D>) -> Tensor<B
 pub fn silu<const D: usize, B: Backend>(tensor: Tensor<B, D>) -> Tensor<B, D> {
     tensor.clone().mul(sigmoid(tensor))
 }
+
+/// Applies the tanh function
+pub fn tanh<const D: usize, B: Backend>(tensor: Tensor<B, D>) -> Tensor<B, D> {
+    tensor.tanh()
+}
```
Comment (antimora) [inline burn-tensor/src/tensor/activation/base.rs:76]: Since this is brand new activation function, we will need a unit test similar to silu: https://github.com/antimora/burn/blob/828ef56cb381b57f0e6581117ba39cff101223a0/burn-tensor/src/tests/activation/silu.rs#L2 

To add a test: 
1. add a file under tests/activation
2. make sure to put a macro with a correct name: #[burn_tensor_testgen::testgen(tanh)] 
3. add to mod.rs like this: https://github.com/antimora/burn/blob/828ef56cb381b57f0e6581117ba39cff101223a0/burn-tensor/src/tests/activation/mod.rs#L4
4. add a module call like this: https://github.com/antimora/burn/blob/828ef56cb381b57f0e6581117ba39cff101223a0/burn-tensor/src/tests/mod.rs#L15


To test the activation function, run `cargo test` under `burn-ndarray` (or other backends). Tensor tests run with a backend.

**Review by antimora — CHANGES_REQUESTED** [review]
Thank you so much for adding this op. There are a couple straight forward additions I am requesting. Please see my inline comments.

File: burn-tensor/src/tensor/activation/base.rs (hunk 2 above)
Comment (AuruTus) [reply burn-tensor/src/tensor/activation/base.rs:76]: Really thank you for the guides. Will complete this part later.

File: burn-import/onnx-tests/tests/tanh/tanh.py (hunk 1 above)
Comment (AuruTus) [reply burn-import/onnx-tests/tests/tanh/tanh.py:18]: Done with it.

File: burn-tensor/src/tensor/activation/base.rs (hunk 2 above)
Comment (AuruTus) [reply burn-tensor/src/tensor/activation/base.rs:76]: Oh I just noticed that there's already a unit test for `tanh` but under `ops` dir. Should we keep this? Or just move it under `activation`, for they actually have the same test data and logic.

Comment (nathanielsimard) [reply burn-tensor/src/tensor/activation/base.rs:76]: I don't think the test is necessary, since it is already tested as a tensor operation. However adding this alias can be more intuitive.

**Review by nathanielsimard — CHANGES_REQUESTED** [review]
I would remove the duplicated test, but other than that LGTM

File: burn-tensor/src/tensor/activation/base.rs (hunk 2 above)
Comment (AuruTus) [reply burn-tensor/src/tensor/activation/base.rs:76]: I think we can keep the new activation test to stay consistent with other activation funcs. It's indeed more intuitive for new maintainers to find calling chain and understand the project struct. And these test snippet are not huge burdens for `tensor` code base. Though the long name and redundancy are really ugly. 😅

Comment (antimora) [reply burn-tensor/src/tensor/activation/base.rs:76]: I would leave it. For any public API, we should always have tests. It just happens that the underlying code/function is already tested but there is no guaranty unless we have tests. Also have tests will ensure confidence if we need to do a refactor.

Comment (AuruTus) [reply burn-tensor/src/tensor/activation/base.rs:76]: Ok it should exist there. We can keep both.

**Review by nathanielsimard — APPROVED** [review]

**Review by antimora — APPROVED** [review]
Looks good. If you have a chance to update the readme file by checking off Tanh, it'd be great. 

Approving ahead.
